internal/server/config.go

```diff
@@ -46,10 +46,58 @@ type KeyProvider struct {
 	Config interface{} // contains secret-provider-specific config
 }
 
-type nativeSecretProviderConfig struct {
+type nativeKeyProviderConfig struct {
 	SecretProvider string `mapstructure:"secretProvider"`
 }
 
+type AWSConfig struct {
+	Endpoint        string `mapstructure:"endpoint" validate:"required"`
+	Region          string `mapstructure:"region" validate:"required"`
+	AccessKeyID     string `mapstructure:"accessKeyID" validate:"required"`
+	SecretAccessKey string `mapstructure:"secretAccessKey" validate:"required"`
+}
+
+type AWSKMSConfig struct {
+	AWSConfig `mapstructure:",squash"`
+
+	EncryptionAlgorithm string `mapstructure:"encryptionAlgorithm"`
+	// aws tags?
+}
+
+type AWSSecretsManagerConfig struct {
+	AWSConfig `mapstructure:",squash"`
+
+	UseSecretMaps bool `mapstructure:"useSecretMaps"`
+}
+
+type AWSSSMConfig struct {
+	AWSConfig `mapstructure:",squash"`
+	KeyID     string `mapstructure:"keyID" validate:"required"` // KMS key to use for decryption
+}
+
+type GenericConfig struct {
+	Base64           bool `mapstructure:"base64"`
+	Base64URLEncoded bool `mapstructure:"base64UrlEncoded"`
+	Base64Raw        bool `mapstructure:"base64Raw"`
+}
+
+type FileConfig struct {
+	GenericConfig `mapstructure:",squash"`
+	Path          string `mapstructure:"path" validate:"required"`
+}
+
+type KubernetesConfig struct {
+	Namespace string `mapstructure:"namespace"`
+}
+
+type VaultConfig struct {
+	TransitMount string `mapstructure:"transitMount"`              // mounting point. defaults to /transit
```

Comment:
Does it still default to `/transit` ? The expected value in the tests changed to empty string.

Reply:
yes but the default comes from the secrets package, not this struct. So it just needs to set the transit only when it's not "" to use the default /transit value. No functionality should have changed.